torch/fx/experimental/symbolic_shapes.py

```diff
@@ -412,6 +412,29 @@ def safe_expand(r):
     'floordiv': lambda a, b: FloorDiv(a, b),
 }
 
+# Drop in replacement for math.floor/ceil.  Actually, math.floor/ceil
+# directly usable, but this has a more relaxed type signature for mypy
+# (mypy requires SupportFloat which is too strict)
+def _sym_floor(x):
+    return sympy.floor(x)  # type: ignore[type]
+
+def _sym_ceil(x):
+    return sympy.ceil(x)  # type: ignore[type]
+
+
+def sym_int_conversion(a):
+    if isinstance(a, sympy.Mul):
+        aa = a.args
+        if len(aa) == 2 and isinstance(aa[0], sympy.Float) and aa[1].is_integer:
+            coef = sympy.Integer(aa[0])
+            print(len(aa), coef)
+            if aa[0] == coef:
+                return (coef * aa[1])
+    # this is wrong, but a > 0 errors out (cannot evaluate truth value)
+    return _sym_floor(a) #if a > 0 else _sym_ceil(a)
```

Comment:
Yeah, doing this entirely in sympy means you can't guard. I'll push a suggested fix